Description: Use debian specific path for nologin executable
Author: Laurent Bigonville <bigon@debian.org>
Forwarded: not-needed
Origin: vendor
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=805086

--- a/src/user-classify.c
+++ b/src/user-classify.c
@@ -79,7 +79,7 @@
         return FALSE;
 }
 
-#define PATH_NOLOGIN "/sbin/nologin"
+#define PATH_NOLOGIN "/usr/sbin/nologin"
 #define PATH_FALSE "/bin/false"
 
 #ifdef ENABLE_USER_HEURISTICS
