Subject: Don't override DefaultStandardOutput.
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=741124
Forwarded: no
---
 data/accounts-daemon.service.in |    1 -
 1 file changed, 1 deletion(-)

--- a/data/accounts-daemon.service.in
+++ b/data/accounts-daemon.service.in
@@ -11,7 +11,6 @@
 Type=dbus
 BusName=org.freedesktop.Accounts
 ExecStart=@libexecdir@/accounts-daemon
-StandardOutput=syslog
 Environment=GVFS_DISABLE_FUSE=1
 Environment=GIO_USE_VFS=local
 Environment=GVFS_REMOTE_VOLUME_MONITOR_IGNORE=1
